src/Microsoft.VisualStudio.ProjectSystem.VisualBasic.VS/ProjectSystem/VS/Automation/VSImports.cs

```diff
@@ -0,0 +1,163 @@
+    internal class VSImports : ConnectionPointContainer,
+                               IEventSource<_dispImportsEvents>,
+                               Imports,
+                               ImportsEvents
+    {
+        private const string importItemTypeName = "Import";
+
+        private readonly ActiveConfiguredProject<ConfiguredProject> _activeConfiguredProject;
+        private readonly IProjectThreadingService _threadingService;
+        private readonly IProjectLockService _lockService;
+        private readonly VSLangProj.VSProject _vsProject;
+
+        public event _dispImportsEvents_ImportAddedEventHandler ImportAdded;
+        public event _dispImportsEvents_ImportRemovedEventHandler ImportRemoved;
+
+        [ImportingConstructor]
+        public VSImports(
+            [Import(ExportContractNames.VsTypes.CpsVSProject)] VSLangProj.VSProject vsProject,
+            IProjectThreadingService threadingService,
+            ActiveConfiguredProject<ConfiguredProject> activeConfiguredProject,
+            IProjectLockService lockService)
+        {
+            Requires.NotNull(vsProject, nameof(vsProject));
+            Requires.NotNull(threadingService, nameof(threadingService));
+            Requires.NotNull(activeConfiguredProject, nameof(activeConfiguredProject));
+            Requires.NotNull(lockService, nameof(lockService));
+
+            _vsProject = vsProject;
+            _activeConfiguredProject = activeConfiguredProject;
+            _lockService = lockService;
+            _threadingService = threadingService;
+
+            AddEventSource(this);
+        }
+
+        private ConfiguredProject ConfiguredProject => _activeConfiguredProject.Value;
+
+        public string Item(int lIndex)
+        {
+            return _threadingService.ExecuteSynchronously(async () =>
+            {
+                using (var access = await _lockService.ReadLockAsync())
+                {
+                    var project = await access.GetProjectAsync(ConfiguredProject).ConfigureAwait(true);
+                    var importsList = project.GetItems(importItemTypeName)
+                                             .Select(p => p.EvaluatedInclude)
+                                             .ToList();
+                    return importsList.ElementAt(lIndex - 1);
+                }
+            });
+        }
+
+        public void Add(string bstrImport)
+        {
+            _threadingService.ExecuteSynchronously(async () =>
+            {
+                using (var access = await _lockService.WriteLockAsync())
+                {
+                    var project = await access.GetProjectAsync(ConfiguredProject).ConfigureAwait(true);
+                    await access.CheckoutAsync(project.Xml.ContainingProject.FullPath).ConfigureAwait(true);
+                    project.AddItem(importItemTypeName, bstrImport);
+                }
+            });
+        }
+
+        public void Remove(object index)
```

Comment:
since the parameter was named index, I assumed it was going to be int but the legacy project system checks to see if `index` is a string as well. I will make those changes